main/src/main/scala/org/clulab/polarity/ml/DeepLearningPolarityClassifier.scala

```diff
@@ -0,0 +1,659 @@
+  val w2v_wemb_lp: LookupParameter = pc.addLookupParameters(w2i.size, /*1579375,*/ Dim(Seq(WEM_DIMENSIONS)))
+  val c2v_cemb:LookupParameter = pc.addLookupParameters(c2i.size, /*1579375,*/ Dim(Seq(CEM_DIMENSIONS)))
+
+  val p_W = pc.addParameters(Dim(1, 2*HIDDEN_SIZE+1))
+  //val p_V = pc.addParameters(Dim(1, MLP_HIDDEN_SIZE))
+  val p_b = pc.addParameters(Dim(1))
+  //val p_bv = pc.addParameters(Dim(1))
+
+
+  val builderFwd = new LstmBuilder(NUM_LAYERS, WEM_DIMENSIONS+CEM_DIMENSIONS*2, HIDDEN_SIZE, pc)
+  val builderBwd = new LstmBuilder(NUM_LAYERS, WEM_DIMENSIONS+CEM_DIMENSIONS*2, HIDDEN_SIZE, pc)
+  val charFwRnnBuilder = new LstmBuilder(NUM_LAYERS, CEM_DIMENSIONS, CEM_DIMENSIONS, pc)
+  val charBwRnnBuilder = new LstmBuilder(NUM_LAYERS, CEM_DIMENSIONS, CEM_DIMENSIONS, pc)
+
+
+  //val sgd = new SimpleSGDTrainer(pc)
+  val sgd = new AdamTrainer(pc)
+  if (maskOption=="tag") {
+    sgd.clipThreshold = 4.0.toFloat
+  }
+
+  val missing_vec = new FloatVector(WEM_DIMENSIONS)
+  val missing_charVec = new FloatVector(CEM_DIMENSIONS*2)
+
+  private var _isFitted = false
+  
+  if (Files.exists(Paths.get(savedModelPath))){
+    logger.info(s"Loading saved model $savedModelPath ...")
+    _isFitted=true
+    val modelLoader = new ModelLoader(savedModelPath)
+    modelLoader.populateModel(pc, "/allParams")
+    logger.info("Loading model finished!")
+  }
+  else{
+    logger.info("Could not find specified model. ")
+  }
+  
+  
+
+  /**
+    * Trains the classifier. This method is meant to have side effects by fitting the parameters
+    *
+    * @param trainingPath Training data
+    * @param trainingRatio Ratio of training samples in the dataset
+    */
+  override def fit(trainingPath:String = spreadsheetPath, trainRatio:Float=0.8.toFloat, saveFlag:Boolean=true): Unit = {
+    
+    if (!_isFitted){
+      val (sens_train, labels_train, sens_test, labels_test) = this.readFromSpreadsheet(trainingPath, trainRatio, maskOption)
+      val N_EPOCH = this.N_EPOCH
+      for (epoch <- 1 to N_EPOCH) {
+        logger.info(s"epoch $epoch")
+        this.fitSingleEpoch(sens_train, labels_train)
+        this.testSingleEpoch(sens_test, labels_test)
+        if (maskOption=="tag") {
+          sgd.learningRate=sgd.learningRate*0.6.toFloat
+        }
+        else{
+          sgd.learningRate=sgd.learningRate*0.3.toFloat
+        }
+      }
+      if (saveFlag) {save()}
+      _isFitted=true
+    }
+
+  }
+
+  /**
+    * Returns whether fit has been called before. Mostly for control
+    *
+    * @return True if the parameters of the model have been fit previously
+    */
+  override def isFitted: Boolean = _isFitted
+
+  /**
+    * Gets the predicted polarity for the provided argument
+    *
+    * @param events Ordered sequence of events to get their polarity from
+    * @return Predictions of Polarity subclasses
+    */
+  override def predict(events: Seq[BioEventMention]): Seq[Polarity] = {
+    var predictions = Seq[Polarity]()
```

Comment:
Please turn this into an ArrayBuffer[Polarity] as we discussed.

Reply:
Done.